shinysdr/plugins/flightradar24/__init__.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2013, 2014, 2015, 2016, 2017, 2018 Kevin Reid and the ShinySDR contributors
+#
+# This file is part of ShinySDR.
+# 
+# ShinySDR is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# ShinySDR is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with ShinySDR.  If not, see <http://www.gnu.org/licenses/>.
+
+# pylint: disable=maybe-no-member, no-member
+# (maybe-no-member: GR swig)
+# (no-member: Twisted reactor)
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os.path
+
+import six
+
+from twisted.internet import task
+from twisted.python.url import URL
+from twisted.web import static
+from twisted.web.client import Agent, readBody
+from zope.interface import Interface, implementer
+
+from shinysdr.devices import Device, IComponent, ITopWatcher
+from shinysdr.interfaces import ClientResourceDef
+from shinysdr.telemetry import ITelemetryMessage, ITelemetryObject, TelemetryItem, Track, empty_track
+from shinysdr.types import TimestampT
+from shinysdr.values import ExportedState, exported_value, setter
+
+_POLLING_INTERVAL = 8
+drop_unheard_timeout_seconds = 60
+
+
+_SECONDS_PER_HOUR = 60 * 60
+_METERS_PER_NAUTICAL_MILE = 1852
+_KNOTS_TO_METERS_PER_SECOND = _METERS_PER_NAUTICAL_MILE / _SECONDS_PER_HOUR
+_CM_PER_INCH = 2.54
+_INCH_PER_FOOT = 12
+_METERS_PER_FEET = (_CM_PER_INCH * _INCH_PER_FOOT) / 100
+_FEET_PER_MINUTE_TO_METERS_PER_SECOND = _METERS_PER_FEET * 60
+
+
+# TODO: This really shouldn't be a Device, but that's the only way right now to hook into the config.
```

Comment:
Actually, being a Device is fine and correct, since it connects to something in the outside world and receives information. Rather, we should improve support for non-RF devices. (Delete this comment.)